fcrepo-http-api/src/test/java/org/fcrepo/http/api/ExternalContentPathValidatorTest.java

```diff
@@ -0,0 +1,315 @@
+        final String path2 = "http://example.com/file";
+        validator.validate(path2);
+        final String path3 = "https://example.com/file";
+        validator.validate(path3);
+    }
+
+    @Test
+    public void testCaseInsensitive() throws Exception {
+        final String goodPath = "FILE://" + dataDir.getAbsolutePath() + "/";
+        final String extPath = dataUri + "/file.txt";
+
+        addAllowedPath(goodPath);
+
+        validator.validate(extPath);
+    }
+
+    @Test
+    public void testFileSlashes() throws Exception {
+        final File oneFolder = tmpDir.newFolder("one");
+        final File twoFolder = tmpDir.newFolder("two");
+        final File threeFolder = tmpDir.newFolder("three");
+        final File manyFolder = tmpDir.newFolder("toomany");
+
+        addAllowedPath("file:" + oneFolder.getAbsolutePath() + "/");
+        addAllowedPath("file:/" + twoFolder.getAbsolutePath() + "/");
+        addAllowedPath("file://" + threeFolder.getAbsolutePath() + "/");
+        addAllowedPath("file:///" + manyFolder.getAbsolutePath() + "/");
+
+        validator.validate("file:" + oneFolder.getAbsolutePath() + "/file");
+        validator.validate("file:/" + oneFolder.getAbsolutePath() + "/file");
+        validator.validate("file://" + oneFolder.getAbsolutePath() + "/file");
+
+        validator.validate("file:" + twoFolder.getAbsolutePath() + "/file");
+        validator.validate("file:/" + twoFolder.getAbsolutePath() + "/file");
+        validator.validate("file://" + twoFolder.getAbsolutePath() + "/file");
+
+        validator.validate("file:" + threeFolder.getAbsolutePath() + "/file");
+        validator.validate("file:/" + threeFolder.getAbsolutePath() + "/file");
+        validator.validate("file://" + threeFolder.getAbsolutePath() + "/file");
+
+        try {
+            validator.validate("file:" + manyFolder.getAbsolutePath() + "file");
+            fail();
+        } catch (final ExternalMessageBodyException e) {
+            // expected
+        }
+    }
+
+    @Test(expected = ExternalMessageBodyException.class)
+    public void testDisallowDirectoryUriWithoutSlash() throws Exception {
+        final String allowDir = dataUri.substring(0, dataUri.length() - 1);
+        addAllowedPath(allowDir);
+
+        final String extPath = dataUri + "file.txt";
+        validator.validate(extPath);
+    }
+
+    @Test(expected = ExternalMessageBodyException.class)
+    public void testDisallowFileUriWithSlash() throws Exception {
+        new File(dataDir, "file").createNewFile();
+        final String extPath = dataUri + "file/";
+        final String allowExactPath = extPath + "/";
+
+        addAllowedPath(allowExactPath);
+
+        validator.validate(extPath);
+    }
+
+    /*
+     * Test ignored because it takes around 10+ seconds to poll for events on MacOS:
+     * https://bugs.openjdk.java.net/browse/JDK-7133447 Can be enabled for one off testing
+     */
+    @Ignore("Test is ignored due to file event timing")
```

Comment:
Do we want to make a note (JIRA) to come back to this?  Or... what are the plans for this?

Reply:
Its just ignored because the test is kind of long. We could turn it on and it should work, just wasn't sure if it was worth 10-20s in every build (at least for people on macs, its probably faster in other environments). Seemed worth keeping around since its the only test of the reloading mechanism rather than deleting before making my PR

Reply:
Ah, okay. I read it as not working, but I totally see what you mean now.   My bad.